mlrun/datastore/sources.py

```
@@ -412,6 +424,92 @@ def to_dataframe(
+                    if value:
+                        if op.lower() == "in":
+                            new_filter = (
+                                col(col_name).isin(value) | col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter | isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                        else:
+                            new_filter = (
+                                ~col(col_name).isin(value) & ~col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter & ~isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                    else:
+                        if op.lower() == "in":
+                            new_filter = col(col_name).isNull()
+                            new_filter = (
+                                new_filter & isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                        else:
+                            new_filter = ~col(col_name).isNull() & ~isnan(col(col_name))
+                            new_filter = (
+                                new_filter & ~isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                else:
+                    if op.lower() == "in":
+                        new_filter = col(col_name).isin(value)
+                    elif op.lower() == "not in":
+                        new_filter = ~col(col_name).isin(value)
+            else:
+                new_filter = operators[op](col(col_name), value)
```

Should probably raise a better error on invalid operator.

---

This issue is still outstanding.

---

plz look at else case:

```
  elif op in operators:
      new_filter = operators[op](col(col_name), value)
  else:
      raise mlrun.errors.MLRunInvalidArgumentError(
          f"additional filter operator is not supported,"
          f"given Operator: {op} ."
      )
```